proto/servicemanager/v4/servicemanager.proto

```diff
@@ -0,0 +1,334 @@
+syntax = "proto3";
+
+package servicemanager.v3;
+
+import "google/protobuf/timestamp.proto";
+import "common/v1/common.proto";
+
+/***********************************************************************************************************************
+ * SMService
+ **********************************************************************************************************************/
+
+service SMService {
+    rpc RegisterSM(stream SMOutgoingMessages) returns (stream SMIncomingMessages) {}
+}
+
+message SMIncomingMessages {
+    oneof SMIncomingMessage {
+        GetUnitConfigStatus     get_unit_config_status     = 1;
+        CheckUnitConfig         check_unit_config          = 2;
+        SetUnitConfig           set_unit_config            = 3;
+        RunInstances            run_instances              = 4;
+        OverrideEnvVars         override_env_vars          = 5;
+        SystemLogRequest        system_log_request         = 6;
+        InstanceLogRequest      instance_log_request       = 7;
+        InstanceCrashLogRequest instance_crash_log_request = 8;
+        GetNodeMonitoring       get_node_monitoring        = 9;
+        ConnectionStatus        connection_status          = 10;
+        ImageContentInfo        image_content_info         = 11;
+        ImageContent            image_content              = 12;
+        UpdateNetworks          update_networks            = 13;
+        ClockSync               clock_sync                 = 14;
+    }
+}
+
+message GetUnitConfigStatus {}
+
+message CheckUnitConfig {
+    string unit_config    = 1;
+    string vendor_version = 2;
+}
+
+message SetUnitConfig {
+    string unit_config    = 1;
+    string vendor_version = 2;
+}
+
+message UpdateNetworks {
+    repeated NetworkParameters networks = 1;
+}
+
+message ClockSync {
+    google.protobuf.Timestamp current_time = 1;
+}
+
+message RunInstances {
+    repeated ServiceInfo  services      = 1;
+    repeated LayerInfo    layers        = 2;
+    repeated InstanceInfo instances     = 3;
+    bool                  force_restart = 4;
+}
+
+message ServiceInfo {
+    VersionInfo version_info = 1;
+    string      url          = 2;
+    string      service_id   = 3;
+    string      provider_id  = 4;
+    uint32      gid          = 5;
+    bytes       sha256       = 6;
+    bytes       sha512       = 7;
+    uint64      size         = 8;
+}
+
+message LayerInfo {
+    VersionInfo version_info = 1;
+    string      url          = 2;
+    string      layer_id     = 3;
+    string      digest       = 4;
```

Comment:
shouldnt digest be byte array?

Reply:
Digest is string for layer everywhere in Aos code. Don't want to change it.